org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -55,14 +59,18 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
       String useNeighborhoodStr = options.getString("useNeighborhood");
       useNeighborhood = Boolean.parseBoolean(useNeighborhoodStr);
     }
-
-    final String executionTypeString = lambdaInput.get("executionType");
-    // TODO: in order to not break current behavior in Javalab, execution type defaults to 'RUN'.
-    // Once Javalab is sending the execution type parameter, remove this fallback
-    final ExecutionType executionType =
-        executionTypeString == null
-            ? ExecutionType.RUN
-            : ExecutionType.valueOf(executionTypeString);
+    final List<String> compileList;
+    if (options.has("compileList")) {
+      compileList =
```

Comment:
could we simplify this logic by sending the compile list in a different format? Also it might be worth creating a helper method for this code since it's duplicated across the websocket server and the lambda request handler.

Reply:
Hmm yeah, the logic here is just to convert the JSONArray into a list of Strings - is there a format that would be more friendly for conversion? Agreed on duplicated code - I'll pull this out into a helper.

Reply:
yeah this might be the simplest option then